### PR TITLE
Preventing multiple AI from being created

### PR DESCRIPTION
**Pull request by Nordsoft91**
1) Server hangs and crashes if press "A" many times during battle
This PR makes it stable.
2) Fix crash when pressing "V" during AI turn

### CODE REVIEW

File: client/mapHandler.cpp [hunk 1]
```diff
@@ -536,6 +536,8 @@ void CMapHandler::CMapWorldViewBlitter::drawTileOverlay(SDL_Surface * targetSurf
 	for(auto & object : objects)
 	{
 		const CGObjectInstance * obj = object.obj;
+		if(!obj)
```
Comment (nullkiller) [inline client/mapHandler.cpp:539]: Interesting case. What does it mean?

Comment (Nordsoft91) [reply client/mapHandler.cpp:539]: Multithreading problem - attempt to dereference object which was deleted. General rule - all ptrs has to be checked for nullptr. If it's not expected, we should use references

Comment (nullkiller) [reply client/mapHandler.cpp:539]: Usually multithreading problems are solved by proper locking. Any additional details on bug mechanics. Should not object be erased first from objects list and deleted afterwards?

Comment (Nordsoft91) [reply client/mapHandler.cpp:539]: Why it should be erased from the list? List is valid, but object has an obj field which is a pointer and could be nullptr, there is no restrictions

Comment (Nordsoft91) [reply client/mapHandler.cpp:539]: I agree that maybe root cause is in another place, but rendering must not crash the game

**Pull request merged**